Kudu.Services/Jobs/JobsController.cs

```diff
@@ -178,11 +178,12 @@ public HttpResponseMessage InvokeTriggeredJob(string jobName, string arguments =
         {
             try
             {
-                _triggeredJobsManager.InvokeTriggeredJob(jobName, arguments, "External - " + Request.Headers.UserAgent);
+                string runId = _triggeredJobsManager.InvokeTriggeredJob(jobName, arguments, "External - " + Request.Headers.UserAgent);
 
                 // Return a 200 in the ARM case, otherwise a 202 can cause it to poll on /run, which we don't support
                 // For non-ARM, stay with the 202 to reduce potential impact of change
-                return Request.CreateResponse(ArmUtils.IsArmRequest(Request) ? HttpStatusCode.OK : HttpStatusCode.Accepted);
+                return Request.CreateResponse(ArmUtils.IsArmRequest(Request) ? HttpStatusCode.OK : HttpStatusCode.Accepted,
+                    ArmUtils.AddEnvelopeOnArmRequest(_triggeredJobsManager.GetJobRun(jobName, runId), Request));
```

Comment:
that would do.  Or if you want to avoid touching file system again, the location header representing a link url to specific run history would suffice.   This is what ARM async operation usually does.